lib/market.rb

```diff
@@ -0,0 +1,58 @@
+class Market
+  attr_reader :name, :vendors
+
+  def initialize(name)
+    @name = name
+    @vendors = []
+  end
+
+  def add_vendor(vendor)
+    @vendors << vendor
+  end
+
+  def vendor_names
+    @vendors.map do |vendor|
+    vendor.name
+    end
+  end
+
+  def vendors_that_sell(item)
+    @vendors.select do |vendor|
+      vendor.inventory[item] > 0
+    end
+  end
+
+  def total_inventory
+    inventory = Hash.new do |hash, key|
+      hash[key] = {quantity: 0, vendors: []}
+    end
+
+    @vendors.each do |vendor|
```

Comment:
Think about how you could do this without each - I challenge you to look into #reduce